Remove formula from the Scaling Spells

theriou opened the pull request:
I know you did not like adding a Formula to the spell data that is displayed, so this is a suggested change of using Base value then just a simple note after that it (Scales to Item/PC Level), so for the Rune it would just list something like "Absorb Damage: 100%, Total: 200 (Scales to Item/PC Level)" then people can scale it out themselves, if you don't wish to expose the formula itself

Re-added the Max to Spell Illusions, was it causing an issue with some? It is currently how the ones with Max values get them (such as Duende Illusion and others in TBL)

theriou committed:
Remove formula from the Scaling Spells

And just use base value then make a note that this scales based on an Items or PC's level that uses it, re add Max to Illusion for those specific Illusions

theriou committed:
Added v to 340, 374, 469, 470

to help differentiate their SPA Versions since display wise they do the "same" thing, but functionally they are different (one is only 1 line chosen per spell, other is chance per spell line in a spell, such that it can do all)

rumstil commented:
Thanks Riou. I'm still hesitant to do this given there are a whole lot of level based calcs for old spells (e.g. pretty much all of 100..143 are level calcs). I'll leave it for now, but I may revert it in the future.

As for the illusion changes I reverted, they broke how some spells are displayed. e.g.:

[598] Illusion: Fire Elemental
1: Illusion: Elemental (1)

Instead of:

1: Illusion: Fire Elemental

theriou committed:
Fix Illusions

A lot of illusions have a Max of 3 for default values when the race has potentially no gender?

theriou commented:
They shouldn't touch the level scaling stuff since it should only be applied to those specific Calc numbers, for example on SPA 0, only calcs 2060, 2070, 2090, 2100 should show the base value with the change, only 3 effects don't show how much it scales by (2100 calc spa 0) these 3 effects should/do scale it just might need another change in "display"

Everything on SPA 55 and 92 with those specific Calcs are directly used in direct Scaling since the Proc consolidation with TBL


Maybe another alternative is to list some that differ such as Lay on Hands (id 13546) like 
"Increase Current Hit Points by 1 (Scale: 100 per Item/PC Level)"
this would make the other ones list similar, I dunno, depends on how you want to present it
if like this line is ok it should be like " (Scale: " + (calc - 2000) + " per Item/PC Level)"



For the Illusions, I suppose it's because a lot of illusions use Max == 3 (631 of them), I've put another commit that should fix most/all of the ones that became broken, it seems to use a default of 3 if the specific race has no gender or only 1 specific gender maybe (such as Sirens exclusively being Female or like Golems/Skeletons, etc)?

theriou commented:
To add more fully to the scaling consolidated procs, the following ID's are used for these specific calc displays on the SPA itself that are effected by it

SPA 0
Calc 2060: 46873, 46874, 46877
Calc 2070: 46875, 46876
Calc 2090: 46871, 46872
Calc 2100: 13546, 22091, 32192, 46861 to 46888

Spa 55
Calc 2200: 46883, 46884

Spa 92
Calc 2400: 46873, 46874
Calc 2800: 46885, 46886

Most of these are the item procs that were consolidated, 3 things that weren't do get included in these (1 rogue poison, paladin rank 1 LoH and a non player ability)

I don't think it needs to be on the few other things that can scale outside of these, it should mostly just to better give context to what people see in game vs on Parser since these specific combos currently can/do have an Item Level tied to it that can differs vastly from PC Max Level, while the Calcs for example 100-143 range are more clearly based on Player Level and the Parser expects it to be seen at Max Level

(though maybe the solution to that specific line of thought is to allow a custom Max Level field to calc against for TLP or something, if care about that)
